modules.d/40network/net-lib.sh

```diff
@@ -14,20 +14,20 @@ is_ip() {
 
 get_ip() {
     local iface="$1" ip=""
-    ip=$(ip -o -f inet addr show $iface)
+    ip=$(ip -f inet addr show $iface)
     ip=${ip%%/*}
     ip=${ip##* }
     echo $ip
 }
 
 iface_for_remote_addr() {
-    set -- $(ip -o route get to $1)
-    echo $5
+    set -- $(ip route get to $1 | sed -n 1p)
+    echo ${5}
 }
```

Comment:
Hmm... 
```
$ set -- $(ip -o route get to 127.0.0.1)
$ echo $5
src
```
Should be $4 ?? right?

Reply:
And I think `$()` eliminates all `\n` anyway...

Reply:
It differs, if you resolve a remote ip it's $5, if the ip is from a local link it's $4. The function is not called anywhere(?), but I could use some sed to handle both cases.

Otherwise, I agree with the whitespace issues. I thought some shell might make problems which required the -oneline switch in the first place, thus I kept it conservative.

Typing this from a mobile now, will do a revised patch in a few days

Reply:
Solved in v2 by cutting away everything before and including 'dev'.